cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,260 @@
+
+                    let from = *legacy_or_1559.from().expect("no sender");
+                    match foundry_utils::next_nonce(from, &fork_url, None) {
+                        Ok(nonce) => {
+                            let tx_nonce = *legacy_or_1559.nonce().expect("no nonce");
+                            let offset = if self.force_resume {
+                                match nonce_offset.get(&from) {
+                                    Some(offset) => *offset,
+                                    None => {
+                                        let offset = nonce - tx_nonce;
+                                        nonce_offset.insert(from, offset);
+                                        offset
+                                    }
+                                }
+                            } else {
+                                U256::from(0u32)
+                            };
+
+                            if nonce != tx_nonce + offset {
+                                deployment_sequence
+                                    .save()
+                                    .expect("not able to save deployment sequence");
+                                eyre::bail!(
+                                    "EOA nonce changed unexpectedly while sending transactions."
+                                );
+                            } else if !offset.is_zero() {
+                                legacy_or_1559.set_nonce(tx_nonce + offset);
+                            }
+                        }
+                        Err(_) => {
+                            deployment_sequence
+                                .save()
+                                .expect("not able to save deployment sequence");
+                            eyre::bail!("Not able to query the EOA nonce.");
+                        }
+                    }
+
+                    async fn send<T, U>(
+                        signer: SignerMiddleware<T, U>,
+                        legacy_or_1559: TypedTransaction,
+                    ) -> eyre::Result<Option<TransactionReceipt>>
+                    where
+                        SignerMiddleware<T, U>: Middleware,
+                    {
+                        tracing::debug!("sending transaction: {:?}", legacy_or_1559);
+                        match signer.send_transaction(legacy_or_1559, None).await {
+                            Ok(pending) => pending.await.map_err(|e| eyre::eyre!(e)),
+                            Err(e) => Err(eyre::eyre!(e.to_string())),
+                        }
+                    }
+
+                    let receipt = match send(signer, legacy_or_1559).await {
+                        Ok(Some(res)) => {
+                            let tx_str =
+                                serde_json::to_string_pretty(&res).expect("Bad serialization");
+                            println!("{}", tx_str);
+                            res
+                        }
```

Comment:
Do we want this? It will proceed to wait for each transaction to get mined, whereas ideally we submit all transactions one after the other, and join on their receipts?

Reply:
If there are multiple signers and they "interact" in some way, i don't think it's wise to expect everything to go smoothly.

S1 calls C1
S2 calls C2
Anyone calls C3 which depends on C1 and C2

Maybe can check, that **if there is only one signer**, then we submit all, otherwise sequentially?

Reply:
I like checking if there's only 1 signer and submitting everything together, yep let's go with that.